include/RAJA/policy/openmp/forall.hpp

```diff
@@ -68,52 +67,150 @@ RAJA_INLINE void forall_impl(const omp_parallel_exec<InnerPolicy>&,
   });
 }
 
+
 ///
-/// OpenMP for nowait policy implementation
+/// OpenMP parallel for schedule policy implementation
 ///
 
-template <typename Iterable, typename Func>
-RAJA_INLINE void forall_impl(const omp_for_nowait_exec&,
-                             Iterable&& iter,
-                             Func&& loop_body)
+namespace internal
 {
-  RAJA_EXTRACT_BED_IT(iter);
-#pragma omp for nowait
-  for (decltype(distance_it) i = 0; i < distance_it; ++i) {
-    loop_body(begin_it[i]);
+
+  /// Tag dispatch for omp forall
+
+  template <typename Iterable, typename Func, unsigned int ChunkSize>
+  RAJA_INLINE void forall_impl(const ::RAJA::policy::omp::Auto&,
+                               Iterable&& iter,
+                               Func&& loop_body)
+  {
+    RAJA_EXTRACT_BED_IT(iter);
+    #pragma omp for
+    for (decltype(distance_it) i = 0; i < distance_it; ++i) {
+      loop_body(begin_it[i]);
+    }
   }
-}
 
-///
-/// OpenMP parallel for policy implementation
-///
+  template <typename Iterable, typename Func, unsigned int ChunkSize>
+  RAJA_INLINE void forall_impl(const ::RAJA::policy::omp::Static<ChunkSize>&,
+                               Iterable&& iter,
+                               Func&& loop_body)
+  {
+    RAJA_EXTRACT_BED_IT(iter);
+    #pragma omp for schedule(static, ChunkSize)
+    for (decltype(distance_it) i = 0; i < distance_it; ++i) {
+      loop_body(begin_it[i]);
+    }
+  }
 
-template <typename Iterable, typename Func>
-RAJA_INLINE void forall_impl(const omp_for_exec&,
+  template <typename Iterable, typename Func>
+  RAJA_INLINE void forall_impl(const ::RAJA::policy::omp::Runtime&,
+                               Iterable&& iter,
+                               Func&& loop_body)
+  {
+    RAJA_EXTRACT_BED_IT(iter);
+    #pragma omp for schedule(runtime)
+    for (decltype(distance_it) i = 0; i < distance_it; ++i) {
+      loop_body(begin_it[i]);
+    }
+  }
+
+  // dynamic & guided
+  template <typename Policy, typename Iterable, typename Func, unsigned int ChunkSize>
+  RAJA_INLINE void forall_impl(const Policy&,
+                               Iterable&& iter,
+                               Func&& loop_body)
+  {
+    omp_sched_t prev_sched;
+    int prev_chunk;
+    omp_get_schedule(&prev_sched, &prev_chunk);
```

Comment:
@trws is preserving the old schedule settings important? I have this added in and don't think it should add too much to the implementation (only used for `Guided` and `Dynamic`)

Reply:
As long as it only exists for guided and dynamic I have no issues with it at all.  The dynamic code paths tend to have a good deal more overhead for startup anyway, and are less frequently used.  There may be an implementation at some point where we decide to care, but none exists today that I know of.